openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -181,6 +419,69 @@ def set_smirks(self, smirks):
     # For example, if we have a parameter with units energy/distance**2, can we check to make
     # sure the dimensionality is preserved when the parameter is modified?
 
+    def to_dict(self, return_cosmetic_attributes=False):
+        """
+        Convert this ParameterType-derived object to dict. A unit-bearing attribute ('X') will be converted to two dict
+        entries, one (['X'] containing the unitless value, and another (['X_unit']) containing a string representation
```

Comment:
I think I saw elsewhere in this PR that we can now parse quantities directly from strings using ast? Could this be swapped to use this functionality so only one entry is present?

Reply:
I think @j-wags adopted this solution because (Jeff correct me if I'm wrong) it's more similar to how the force field is serialized in XML format.

Reply:
@andrrizzi is correct here. Per the SMIRNOFF spec, we store units and values separately. Storing each quantity with its units will create a lot of text bloat -- for example, each bond force constant would expand from from `k=20`, with a single `k_unit=kilocalorie_per_mole / angstrom**2` in the section header,  to `k=20 kilocalorie_per_mol / angstrom **2`. Given that we'll have hundreds of these (and probably even more torsion types), this could end up being a significant amount of text.